src/file_operations.rs

```diff
@@ -0,0 +1,97 @@
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+
+enum Op {
+    MkDir(PathBuf),
+    GitInit(PathBuf),
+}
+
+struct FileOperations {
+    root: PathBuf,
+    operations: Vec<Op>,
+}
+
+impl FileOperations {
+    fn rooted_at<P: AsRef<Path>>(path: P) -> FileOperations {
+        FileOperations {
+            root: PathBuf::from(path.as_ref()),
+            operations: vec![],
+        }
+    }
+
+    fn make_dir<P: AsRef<Path>>(&mut self, name: P) {
+        let path = PathBuf::from(name.as_ref());
+        self.operations.push(Op::MkDir(path))
+    }
+
+    fn make_git_repo<P: AsRef<Path>>(&mut self, name: P) {
+        let path = PathBuf::from(name.as_ref());
+        self.operations.push(Op::GitInit(path))
+    }
+
+    fn commit(mut self) -> Vec<io::Result<()>> {
+        let ops = self.operations;
+        self.operations = vec![];
+        self.operations.push(Op::MkDir(PathBuf::from(".")));
```

Comment:
Is `PathBuf::from(".")` intended to reference the current directory? Would [`env::current_dir()`](https://doc.rust-lang.org/1.0.0/std/env/fn.current_dir.html) maybe be better?